packages/components/src/unit-control/styles/unit-control-styles.ts

```diff
@@ -26,21 +26,11 @@ export const Root = styled.div`
 	position: relative;
 `;
 
-const paddingStyles = ( { disableUnits, size }: InputProps ) => {
-	const paddings = {
-		default: {
-			paddingRight: 8,
-		},
-		small: {
-			paddingRight: 8,
-		},
-		'__unstable-large': {
-			paddingRight: disableUnits ? 16 : 8,
-		},
-	};
+const paddingStyles = ( { disableUnits }: InputProps ) => {
+	if ( disableUnits ) return '';
 
 	return css`
-		${ rtl( paddings[ size ] )() };
+		${ rtl( { paddingRight: 8 } )() };
```

Comment:
I simplified the logic here to capture the intent better. Basically, we want the padding-right to always be 8px when units are enabled. Otherwise, it simply inherits from the paddings that `InputControl` specifies.

Reply:
Love this, so much easier to read!

Not for this PR, but I wonder if we should specify these amounts using the space utility (i.e. multiples of `4`) — we should probably do an overhaul of the config before that, though.

Reply:
Yes, good point. I think we're supposed to move to this final design for the unit dropdown eventually, so maybe that would be a good time to clean up 👇 

<img width="158" alt="Final UnitControl design with a larger unit dropdown" src="https://user-images.githubusercontent.com/555336/165744940-f515cb48-ef89-46a5-ab98-456a2b516503.png">